extensions/SemanticScribunto/src/LuaAskResultProcessor.php

```diff
@@ -0,0 +1,215 @@
+	 *
+	 * @return array array ( int|string, array )
+	 */
+	public function getDataFromResultArray( ResultArray $resultArray ) {
+
+		// first, extract the key (label), if any
+		$key = $this->getKeyFromPrintRequest( $resultArray->getPrintRequest() );
+
+		$resultArrayData = [];
+
+		// then get all data that is stored in this printRequest / query field
+		/** @var DataValue $dataValue */
+		while ( ( $dataValue = $resultArray->getNextDataValue() ) !== false ) {
+
+			$resultArrayData[] = $this->getValueFromDataValue( $dataValue );
+		}
+
+		$resultArrayData = $this->extractLuaDataFromDVData( $resultArrayData );
+
+		return [ $key, $resultArrayData ];
+	}
+
+	/**
+	 * Takes an smw query print request and tries to retrieve the label
+	 * falls back to {@see getNumericIndex} if none found
+	 *
+	 * @param PrintRequest $printRequest
+	 *
+	 * @uses getNumericIndex
+	 *
+	 * @return int|string
+	 */
+	public function getKeyFromPrintRequest( PrintRequest $printRequest ) {
+
+		if ( $printRequest->getLabel() !== '' ) {
+			return $printRequest->getText( SMW_OUTPUT_WIKI );
+		}
+
+		return $this->getNumericIndex();
+	}
+
+	/**
+	 * Extracts the data of a single value of the current printRequest / query field
+	 * The value is formatted according to the type of the property
+	 *
+	 * @param DataValue $dataValue
+	 *
+	 * @return mixed
+	 */
+	public function getValueFromDataValue( DataValue $dataValue ) {
+
+		switch ( $dataValue->getTypeID() ) {
+			case '_boo':
+				// boolean value found, convert it
+				$value = in_array( strtolower( $dataValue->getShortText( SMW_OUTPUT_WIKI ) ), $this->msgTrue );
+				break;
+			case '_num':
+				// number value found
+				/** @var \SMWNumberValue $dataValue */
+				$value = $dataValue->getNumber();
+				$value = ( $value == (int)$value ) ? intval( $value ) : $value;
+				break;
+			default:
+				# FIXME ignores parameter link=none|subject
+				$value = $dataValue->getShortText( SMW_OUTPUT_WIKI, new \Linker() );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Takes an array of data fields and returns either null, a single value or a lua table.
+	 *
+	 * @param array $resultArrayData
+	 *
+	 * @return mixed
+	 */
+	public function extractLuaDataFromDVData( $resultArrayData ) {
+
+		if ( empty( $resultArrayData ) ) {
+			// this key has no value(s). set to null
+			$resultArrayData = null;
+		} elseif ( count( $resultArrayData ) == 1 ) {
+			// there was only one semantic value found. reduce the array to this value
+			$resultArrayData = array_shift( $resultArrayData );
+		} else {
+			// $key has multiple values. keep the array
+			// Note: we do not un-shift it (remember: lua array counting starts with 1), but the defer to
+			// conversion to a lua table to a later step
+		}
+
+		return $resultArrayData;
+	}
+
+	/**
+	 * Returns current numeric index (and increases it afterwards)
+	 *
+	 * @uses $numericIndex
+	 *
+	 * @return int
+	 */
+	public function getNumericIndex() {
+		return $this->numericIndex++;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this increment or decrement operator into a dedicated statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS881)